assets/pages/new-contributions-landing/components/StripePaymentRequestButton/StripePaymentRequestButton.jsx

```diff
@@ -156,18 +164,13 @@ function initialisePaymentRequest(props: PropTypes) {
     },
     requestPayerEmail: true,
   });
-  paymentRequest.on('token', ({ complete, token, ...data }) => {
-    // We need to do this so that we can offer marketing permissions on the thank you page
-    updateUserEmail(data, props.updateEmail);
-    const tokenId = props.isTestUser ? 'tok_visa' : token.id;
-    props.onPaymentAuthorised({ paymentMethod: 'Stripe', token: tokenId, stripePaymentMethod: 'StripeApplePay' })
-      .then(onComplete(complete));
-  });
 
   paymentRequest.canMakePayment().then((result) => {
-    if (browserIsCompatible(result)) {
-      trackComponentClick('apple-pay-loaded');
-      props.setCanMakeStripePaymentRequestPayment(true);
+    const paymentMethod = availablePaymentRequestButtonPaymentMethod(result);
+    if (paymentMethod !== null) {
+      trackComponentClick(`${paymentMethod}-loaded`);
+      props.setPaymentRequestButtonPaymentMethod(paymentMethod);
+      setUpPaymentListener(props, paymentRequest, paymentMethod);
```

Comment:
I just wonder if to eliminate race conditions it's worth doing `setUpPaymentListener(props, paymentRequest, paymentMethod);` before `props.setPaymentRequestButtonPaymentMethod(paymentMethod);` since I think the latter might be entirely synchronous (i.e. the state update & DOM re-render all happen before the next line)

it's super-paranoid but I'm just wondering if there's a way someone might manage to click the button before the event handler's bound. could be best just to make that impossible

Reply:
> since I think the latter might be entirely synchronous (i.e. the state update & DOM re-render all happen before the next line)

actually I'm really not sure about that...

Reply:
I think it might be at the React level that the state changes are decoupled from re-renders: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

Reply:
Yeah sure, why not! No harm done